[2023-08-29T01:29:26.698+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: crypto_data_download.fetch_and_store_task manual__2023-08-29T01:29:23.515696+00:00 [queued]>
[2023-08-29T01:29:26.707+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: crypto_data_download.fetch_and_store_task manual__2023-08-29T01:29:23.515696+00:00 [queued]>
[2023-08-29T01:29:26.708+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2023-08-29T01:29:26.722+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): fetch_and_store_task> on 2023-08-29 01:29:23.515696+00:00
[2023-08-29T01:29:26.728+0000] {standard_task_runner.py:57} INFO - Started process 1274 to run task
[2023-08-29T01:29:26.732+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'crypto_data_download', 'fetch_and_store_task', 'manual__2023-08-29T01:29:23.515696+00:00', '--job-id', '137', '--raw', '--subdir', 'DAGS_FOLDER/coin_data_dag.py', '--cfg-path', '/tmp/tmp8w87a30h']
[2023-08-29T01:29:26.736+0000] {standard_task_runner.py:85} INFO - Job 137: Subtask fetch_and_store_task
[2023-08-29T01:29:26.790+0000] {task_command.py:415} INFO - Running <TaskInstance: crypto_data_download.fetch_and_store_task manual__2023-08-29T01:29:23.515696+00:00 [running]> on host 6c9aee5df47e
[2023-08-29T01:29:26.876+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='NicoB' AIRFLOW_CTX_DAG_ID='crypto_data_download' AIRFLOW_CTX_TASK_ID='fetch_and_store_task' AIRFLOW_CTX_EXECUTION_DATE='2023-08-29T01:29:23.515696+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-08-29T01:29:23.515696+00:00'
[2023-08-29T01:29:26.996+0000] {base.py:73} INFO - Using connection ID 'postgres_connection' for task execution.
[2023-08-29T01:29:27.001+0000] {sql.py:418} INFO - Running statement: 
        INSERT INTO coin_data (coin_id, price_usd, date, json_response)
        VALUES (%s, %s, %s, %s);
    , parameters: ('bitcoin', 824.4988000000003, '13-01-2017', '{"id": "bitcoin", "symbol": "btc", "name": "Bitcoin", "localization": {"en": "Bitcoin", "de": "Bitcoin", "es": "Bitcoin", "fr": "Bitcoin", "it": "Bitcoin", "pl": "Bitcoin", "ro": "Bitcoin", "hu": "Bitcoin", "nl": "Bitcoin", "pt": "Bitcoin", "sv": "Bitcoin", "vi": "Bitcoin", "tr": "Bitcoin", "ru": "\\u0411\\u0438\\u0442\\u043a\\u043e\\u0438\\u043d", "ja": "\\u30d3\\u30c3\\u30c8\\u30b3\\u30a4\\u30f3", "zh": "\\u6bd4\\u7279\\u5e01", "zh-tw": "\\u6bd4\\u7279\\u5e63", "ko": "\\ube44\\ud2b8\\ucf54\\uc778", "ar": "\\u0628\\u064a\\u062a\\u0643\\u0648\\u064a\\u0646", "th": "\\u0e1a\\u0e34\\u0e15\\u0e04\\u0e2d\\u0e22\\u0e19\\u0e4c", "id": "Bitcoin", "cs": "Bitcoin", "da": "Bitcoin", "el": "Bitcoin", "hi": "Bitcoin", "no": "Bitcoin", "sk": "Bitcoin", "uk": "Bitcoin", "he": "Bitcoin", "fi": "Bitcoin", "bg": "Bitcoin", "hr": "Bitcoin", "lt": "Bitcoin", "sl": "Bitcoin"}, "image": {"thumb": "https://assets.coingecko.com/coins/images/1/thumb/bitcoin.png?1547033579", "small": "https://assets.coingecko.com/coins/images/1/small/bitcoin.png?1547033579"}, "market_data": {"current_price": {"aed": 3028.398933378401, "ars": 13064.183486000005, "aud": 1100.3921, "bdt": 65430.163521000024, "bhd": 310.90200750400015, "bmd": 824.4988000000003, "brl": 2656.2053000000005, "btc": 1.0, "cad": 1082.2611000000002, "chf": 833.4812, "clp": 543592.0588400002, "cny": 5632.649999999998, "czk": 20934.24055068561, "dkk": 5760.489999999999, "eth": 85.43313528688516, "eur": 775.2239000000001, "gbp": 677.3406999999999, "hkd": 6399.085300000001, "huf": 238045.17104200006, "idr": 10974188.6714, "ils": 3149.581293506001, "inr": 56194.535299999996, "jpy": 94392.73989999999, "krw": 969528.4926, "kwd": 251.9041713712001, "lkr": 123608.86009600003, "ltc": 210.4877813520826, "mmk": 1122101.6418600006, "mxn": 17709.095799999996, "myr": 3681.7861999999996, "ngn": 259189.4427680001, "nok": 7005.551933912003, "nzd": 1156.2992999999997, "php": 41076.5275, "pkr": 86448.69918000003, "pln": 3397.8005, "rub": 49168.93020000002, "sar": 3092.365199280001, "sek": 7344.800000000003, "sgd": 1177.2441000000001, "thb": 29188.90651760001, "try": 3070.017983804801, "twd": 26051.275100000003, "uah": 22765.00468263721, "usd": 824.4988000000003, "vef": 8232.638656973604, "vnd": 18591692.149158508, "xag": 49.06279999999999, "xau": 0.6885000000000001, "xdr": 610.1752000000002, "zar": 11133.3547, "bits": 1000000.0, "sats": 100000000.0}, "market_cap": {"aed": 48758546237.726135, "ars": 210339063172.78345, "aud": 17716793681.347702, "bdt": 1053454225669.559, "bhd": 5005658184.991681, "bmd": 13274790985.975603, "brl": 42766066091.7161, "btc": 16100437.0, "cad": 17424876658.100704, "chf": 13419411551.284399, "clp": 8752069697053.715, "cny": 90688126468.04997, "czk": 337050421129.1589, "dkk": 92746406334.12999, "eth": 1375510812.3989716, "eur": 12481443562.844301, "gbp": 10905481267.885897, "hkd": 103028069730.27612, "huf": 3832631279515.946, "idr": 176689233329989.4, "ils": 50709635192.47188, "inr": 904756575341.926, "jpy": 1519764362017.336, "krw": 15609832414811.266, "kwd": 4055767241.19921, "lkr": 1990156664617.4622, "ltc": 3388945262.928981, "mmk": 18066326792363.496, "mxn": 285124181254.86456, "myr": 59278366760.5694, "ngn": 4173063294351.291, "nok": 112792447562.17836, "nzd": 18616924032.794098, "php": 661350043192.5175, "pkr": 1391861834879.542, "pln": 54706072888.8185, "rub": 791641263042.4977, "sar": 49788431072.0001, "sek": 118254489677.60005, "sgd": 18954144465.6717, "thb": 469954150485.5083, "try": 49428631137.11621, "twd": 419436913517.2187, "uah": 366526523697.5053, "usd": 13274790985.975603, "vef": 132549080040.3681, "vnd": 299334368170921.06, "xag": 789932520.4435998, "xau": 11085150.874500003, "xdr": 9824087366.562403, "zar": 179251875946.0039, "bits": 16100437000000.0, "sats": 1610043700000000}, "total_volume": {"aed": 6832803935.505425, "ars": 29475972717.281395, "aud": 2482751987.7281656, "bdt": 147626349315.97717, "bhd": 701470482.3119261, "bmd": 1860269657.1335685, "brl": 5993044650.528742, "btc": 2256243.01349325, "cad": 2441844045.6505194, "chf": 1880536134.37797, "clp": 1226475784948.1616, "cny": 12708627209.952755, "czk": 47232733985.27148, "dkk": 12997065316.797731, "eth": 192757914.6118583, "eur": 1749093508.2679899, "gbp": 1528245222.1296275, "hkd": 14437891500.872358, "huf": 537087754059.3182, "idr": 24760436518603.023, "ils": 7106220788.901946, "inr": 126788527667.12482, "jpy": 212972959923.86053, "krw": 2187491887811.392, "kwd": 568357026.7260764, "lkr": 278891626997.4646, "ltc": 474911586.1013312, "mmk": 2531733989875.93, "mxn": 39956023674.03265, "myr": 8307004390.925861, "ngn": 584794369416.5087, "nok": 15806227606.553078, "nzd": 2608892217.1321354, "php": 92678628190.43835, "pkr": 195049273550.45465, "pln": 7666263639.368872, "rub": 110937055244.68727, "sar": 6977127376.045162, "sek": 16571653685.505222, "sgd": 2656148775.801149, "thb": 65857266401.8426, "try": 6926706627.258214, "twd": 58778007436.96567, "uah": 51363382767.34131, "usd": 1860269657.1335685, "vef": 18574833452.41114, "vnd": 41947375520556.17, "xag": 110697599.72241662, "xau": 1553423.3147901027, "xdr": 1376703532.0068464, "zar": 25119553758.617237, "bits": 2256243013493.25, "sats": 225624301349325.0}}, "community_data": {"facebook_likes": null, "twitter_followers": 151603, "reddit_average_posts_48h": 2.063, "reddit_average_comments_48h": 117.563, "reddit_subscribers": 202551, "reddit_accounts_active_48h": "664.0"}, "developer_data": {"forks": 7208, "stars": 11025, "subscribers": 1306, "total_issues": 2916, "closed_issues": 2495, "pull_requests_merged": 4142, "pull_request_contributors": 378, "code_additions_deletions_4_weeks": {"additions": null, "deletions": null}, "commit_count_4_weeks": 229}, "public_interest_stats": {"alexa_rank": null, "bing_matches": null}}')
[2023-08-29T01:29:27.003+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/coin_data_dag.py", line 56, in fetch_and_store_crypto_data_for_range
    fetch_and_store_crypto_data_for_date(current_date.strftime('%d-%m-%Y'))
  File "/opt/airflow/dags/coin_data_dag.py", line 51, in fetch_and_store_crypto_data_for_date
    pg_hook.run(insert_query, parameters=values)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 392, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 421, in _run_command
    cur.execute(sql_statement, parameters)
psycopg2.errors.DatetimeFieldOverflow: date/time field value out of range: "13-01-2017"
LINE 3:         VALUES ('bitcoin', 824.4988000000003, '13-01-2017', ...
                                                      ^
HINT:  Perhaps you need a different "datestyle" setting.

[2023-08-29T01:29:27.014+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=crypto_data_download, task_id=fetch_and_store_task, execution_date=20230829T012923, start_date=20230829T012926, end_date=20230829T012927
[2023-08-29T01:29:27.026+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 137 for task fetch_and_store_task (date/time field value out of range: "13-01-2017"
LINE 3:         VALUES ('bitcoin', 824.4988000000003, '13-01-2017', ...
                                                      ^
HINT:  Perhaps you need a different "datestyle" setting.
; 1274)
[2023-08-29T01:29:27.066+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2023-08-29T01:29:27.087+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
